openmetadata-ui/src/main/resources/ui/src/router/AppRouter.tsx

```diff
@@ -114,95 +119,60 @@ const AppRouter = () => {
     }
   }, [location.pathname]);
 
-  return loading ? (
-    <Loader />
-  ) : (
-    <>
-      {isOidcProvider || isAuthenticated ? (
-        <>
-          <AuthenticatedAppRouter />
-          {slackChat}
-        </>
-      ) : (
+  if (loading) {
+    return <Loader />;
+  }
+
+  if (!isAuthenticated && isProtectedRoute(location.pathname)) {
+    return <Redirect to={ROUTES.SIGNIN} />;
+  }
+
+  if (isOidcProvider || isAuthenticated) {
+    return (
+      <>
+        <AuthenticatedAppRouter />
+        {slackChat}
+      </>
+    );
+  }
+
+  return (
+    <Switch>
+      {slackChat}
+      <Route exact component={SigninPage} path={ROUTES.SIGNIN} />
+      {callbackComponent ? (
+        <Route component={callbackComponent} path={ROUTES.CALLBACK} />
+      ) : null}
+
+      <Route exact path={ROUTES.HOME}>
+        {!isAuthenticated && !isSigningIn ? (
+          <>
+            <Redirect to={ROUTES.SIGNIN} />
+          </>
+        ) : (
+          <Redirect to={ROUTES.MY_DATA} />
+        )}
```

Comment:
Shouldn't this will also have the `!isBasicAuthProvider` check?

Reply:
No